src/util/Util.ts

```diff
@@ -51,3 +52,17 @@ export const matchingMediaType = (mediaA: string, mediaB: string): boolean => {
   }
   return subTypeA === subTypeB;
 };
+
+/**
+ * Will pipe two streams.
+ * Makes sure an error of the first stream gets passed to the second.
+ * @param readable - Initial readable stream.
+ * @param destination - The destination for writing data.
+ *
+ * @returns destination as a Readable.
+ */
+export const pipeStreams = (readable: Readable, destination: Duplex): Readable => {
```

Comment:
```suggestion
export const pipeStreamAndErrors = (readable: Readable, destination: Writable): Readable => {
```

Reply:
Destination of type Writable will give a type error. When the destination of a pipe is a Writable, it returns a Writable, not a Readable (see [here](https://nodejs.org/api/stream.html#stream_readable_pipe_destination_options)). Setting the type to Duplex is the solution I stuck with, as it allows a chain of pipes. I coudn't really find another way, but there might be one.

Reply:
Could use a generic for this to be even more flexible:
```typescript
const pipeStreams = <T extends Writable>(readable: Readable, destination: T): T => {
```

Reply:
Makes sense, I'm fine with `Duplex`.

Reply:
Now that I've seen the generic version I actually prefer that one since that would mean we can also use it for both Writable and Duplex streams (and we don't lose the output type anymore).